pyrokinetics/gk_code/GKInputGENE.py

```diff
@@ -123,13 +140,17 @@ def get_local_geometry_miller(self) -> LocalGeometryMiller:
         """
         miller_data = default_miller_inputs()
 
-        for pyro_key, (gene_param, gene_key) in self.pyro_gene_miller.items():
-            miller_data[pyro_key] = self.data[gene_param][gene_key]
+        for (pyro_key, (gene_param, gene_key)), gene_default in zip(
+            self.pyro_gene_miller.items(), self.pyro_gene_miller_default.values()
+        ):
+            miller_data[pyro_key] = self.data[gene_param].get(gene_key, gene_default)
 
-        miller_data["Rmaj"] = self.data["geometry"]["major_r"] / self.data[
+        miller_data["Rmaj"] = self.data["geometry"].get("major_r", 1.0) / self.data[
             "geometry"
         ].get("minor_r", 1.0)
```

Comment:
@dake0795 Does `minor_r` have a default? From the manual I have it seems not so not sure what to do if it isn't given

Reply:
This could potentially lead to bad behaviour here so if it is not defined then this won't work. Will add a todo here